java-processors/src/main/java/com/dataiku/dip/plugins/tradearea/AreaGenerator.java

```diff
@@ -0,0 +1,126 @@
+package com.dataiku.dip.plugins.tradearea;
+
+import com.dataiku.dip.shaker.types.GeoPoint;
+import com.dataiku.dip.utils.DKULogger;
+
+abstract class AreaGenerator {
+    /*
+    An abstract class expected by the TradeAreaProcessor to generate the polygons
+     */
+
+    public abstract String generateArea(GeoPoint.Coords coords);
+
+}
+
+class RectangleAreaGenerator extends AreaGenerator {
+    /*
+    Generates a rectangular area centered on an input geopoint coordinates.
+    The parameters width, height, radius are expected to be in kilometers.
+    If using miles in the processor, those distances must be converted to miles before.
+     */
+
+    double width;
+    double height;
+    double radius;
+
+    private static DKULogger logger = DKULogger.getLogger("dku");
+
+    public RectangleAreaGenerator(double width, double height){
+        if (width < 0 || height < 0){
+            logger.info("Received invalid parameters as input for RectangleAreaGenerator."+" width="+width+" height="+height);
+            // Skip variable assignation if bad input
+        } else {
+            this.width = width;
+            this.height = height;
+            // Compute the radius based on half the width and height
+            this.radius = Math.sqrt(Math.pow(width/2, 2)+Math.pow(height/2, 2));
+        }
+    }
+
+    public String generateArea(GeoPoint.Coords center) {
+        /*
+        Generate a rectangular trade area centered on an input geopoint coordonates.
+
+        Input:
+            GeoPoint.Coords center : Coordinates of the center geopoint
+        Output:
+            The rectangular trade area expressed as a WKT polygon (string)
+            example: `POLYGON((long1 lat1,long2 lat2, ...))`
+         */
+        if (this.radius<=0 || this.width<=0 || this.height<=0){
+            logger.info("Detected invalid input parameter. Distance input parameters should be greater than zero.");
+            return null;
+        }
+        GeoPoint.Coords initCoords = null;
+        // Declaration of the final result
+        StringBuilder str = new StringBuilder();
+        str.append("POLYGON((");
+        // Computation of the diagonal angle of the rectangle (Must be in degree and Math.atan is radian)
+        double diagonalAngle = Math.atan(this.height/this.width)*180/Math.PI;
+        double[] angles = {90-diagonalAngle, 90+diagonalAngle, 270-diagonalAngle, 270+diagonalAngle};
+        // Compute the latitude longitude of the four corners and fill the polygon String
+        for (int i = 0; i < angles.length; i++){
+            double angle;
+            angle = angles[i];
+            GeoPoint.Coords tmpCoords;
+            tmpCoords = GeoUtils.computeDestinationPoint(center.latitude, center.longitude, angle, this.radius);
+            str.append(tmpCoords.longitude + " " + tmpCoords.latitude + ",");
```

Comment:
Avoid temporary string creation by using multiple append.
```suggestion
str.append(tmpCoords.longitude);
str.append(" ");
str.append(tmpCoords.latitude);
str.append(",");
```

Reply:
Ok now